moha/hamiltonians.py

```diff
@@ -703,3 +707,170 @@ def __init__(self,
+            \end{align}
+
+        """
+        # Default charges to an array of ones if not provided
+        if charges is None:
+            charges = np.ones(len(connectivity))
+
+        # Initialize the PPP part
+        self.ocupation_part = HamPPP(connectivity=connectivity,
+                                     alpha=alpha,
+                                     beta=beta,
+                                     u_onsite=u_onsite,
+                                     gamma=gamma,
+                                     charges=charges,
+                                     sym=sym,
+                                     atom_dictionary=atom_dictionary,
+                                     bond_dictionary=bond_dictionary,
+                                     orbital_overlap=orbital_overlap,
+                                     affinity_dct=affinity_dct,
+                                     Rxy_list=Rxy_list)
+
+        connectivity_matrix = np.asarray(
+            self.ocupation_part.connectivity_matrix.todense())
+
+        mu = np.zeros(connectivity_matrix.shape[0])
+
+        # Initialize the Heisenberg part
+        self.spin_part = HamHeisenberg(mu=mu,
+                                       J_eq=J_eq,
+                                       J_ax=J_ax,
+                                       connectivity=connectivity_matrix)
+
+    def generate_zero_body_integral(self):
+        r"""Generate zero body integral.
+
+        Returns
+        -------
+        float
+        """
+        self.zero_energy = self.ocupation_part.generate_zero_body_integral(
+        ) + self.spin_part.generate_zero_body_integral()
+        return self.zero_energy
+
+    def generate_one_body_integral(self, basis: str, dense: bool):
+        r"""
+        Generate one body integral in spatial or spin orbital basis.
+
+        Parameters
+        ----------
+        basis: str
+            ['spatial', 'spin orbital']
+        dense: bool
+            Dense or sparse matrix; default False
+
+        Returns
+        -------
+        scipy.sparse.csr_matrix or np.ndarray
+        """
+        one_body_ppp = self.ocupation_part.generate_one_body_integral(
+            basis, dense)
+        one_body_heisenberg = self.spin_part.generate_one_body_integral(
+            dense, basis)
+        self.one_body = one_body_ppp + one_body_heisenberg
+        return self.one_body
+
+    def generate_two_body_integral(self, basis: str, dense: bool, sym=1):
+        r"""
+        Generate two body integral in spatial or spin orbital basis.
+
+        Parameters
+        ----------
+        basis: str
+            ['spatial', 'spin orbital']
+        dense: bool
+            Dense or sparse matrix; default False
+        sym: int
+            Symmetry -- [2, 4, 8] default is 1
+
+        Returns
+        -------
+        scipy.sparse.csr_matrix or np.ndarray
+        """
+        two_body_ppp = self.ocupation_part.generate_two_body_integral(
+            basis, dense, sym)
```

Comment:
It looks like order of parameters for generating two body hamiltonian for PPP and Heisenberg hamiltonians are inconsistent. We need to change order of Heisenberg model parameters and then change it in this line